go/libraries/doltcore/sqle/dfunctions/dolt_conflicts_resolve.go

```diff
@@ -0,0 +1,496 @@
+}
+
+func (d DoltConflictsCatFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	args, err := getDoltArgs(ctx, row, d.Children())
+	if err != nil {
+		return 1, err
+	}
+	return DoDoltConflictsResolve(ctx, args)
+}
+
+func getProllyRowMaps(ctx *sql.Context, vrw types.ValueReadWriter, ns tree.NodeStore, hash hash.Hash, tblName string) (prolly.Map, error) {
+	rootVal, err := doltdb.LoadRootValueFromRootIshAddr(ctx, vrw, ns, hash)
+	tbl, ok, err := rootVal.GetTable(ctx, tblName)
+	if err != nil {
+		return prolly.Map{}, err
+	}
+	if !ok {
+		return prolly.Map{}, doltdb.ErrTableNotFound
+	}
+
+	idx, err := tbl.GetRowData(ctx)
+	if err != nil {
+		return prolly.Map{}, err
+	}
+
+	return durable.ProllyMapFromIndex(idx), nil
+}
+
+func resolveProllyConflicts(ctx *sql.Context, tbl *doltdb.Table, tblName string, sch schema.Schema) (*doltdb.Table, error) {
+	var err error
+	artifactIdx, err := tbl.GetArtifacts(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	artifactMap := durable.ProllyMapFromArtifactIndex(artifactIdx)
+	iter, err := artifactMap.IterAllConflicts(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutable prolly map
+	ourIdx, err := tbl.GetRowData(ctx)
+	if err != nil {
+		return nil, err
+	}
+	ourMap := durable.ProllyMapFromIndex(ourIdx)
+	mutMap := ourMap.Mutate()
+
+	// accumulate index edits
+	once := true
+	var theirMap prolly.Map
+	var indexEdits []tree.Diff
+	for {
+		cnfArt, err := iter.Next(ctx)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		if once {
+			theirMap, err = getProllyRowMaps(ctx, tbl.ValueReadWriter(), tbl.NodeStore(), cnfArt.TheirRootIsh, tblName)
+			if err != nil {
+				return nil, err
+			}
+			once = false
+		}
+
+		var ourRow, theirRow val.Tuple
+		err = ourMap.Get(ctx, cnfArt.Key, func(_, v val.Tuple) error {
+			ourRow = v
+			return nil
+		})
+		if err != nil {
+			return nil, err
+		}
+		err = theirMap.Get(ctx, cnfArt.Key, func(_, v val.Tuple) error {
+			theirRow = v
+			return nil
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		edit := tree.Diff{
+			Key:  tree.Item(cnfArt.Key),
+			From: tree.Item(ourRow),
+			To:   tree.Item(theirRow),
+		}
+		indexEdits = append(indexEdits, edit)
+
+		if theirRow == nil {
+			err = mutMap.Delete(ctx, cnfArt.Key)
+		} else {
+			err = mutMap.Put(ctx, cnfArt.Key, theirRow)
+		}
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Update table
+	newMap, err := mutMap.Map(ctx)
+	if err != nil {
+		return nil, err
+	}
+	newIdx := durable.IndexFromProllyMap(newMap)
+	newTbl, err := tbl.UpdateRows(ctx, newIdx)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutable secondary indexes
+	idxSet, err := tbl.GetIndexSet(ctx)
+	if err != nil {
+		return nil, err
+	}
+	mutIdxs, err := merge.GetMutableSecondaryIdxs(ctx, sch, idxSet)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: could do this inside the loop while resolving row data
+	// Update secondary indexes
+	for _, mutIdx := range mutIdxs {
```

Comment:
yea do this inside the conflict iteration loop, not good to hold the index edits in memory. Also, I wouldn't write code to load the tuples into a tree.Diff only to check if `From` or `To` is empty.